src/PageLayout/PageLayout.tsx

```diff
@@ -0,0 +1,333 @@
+import React from 'react'
+import {Box} from '..'
+import {BetterSystemStyleObject, merge, SxProp} from '../sx'
+
+const REGION_ORDER = {
+  header: 0,
+  paneStart: 1,
+  content: 2,
+  paneEnd: 3,
+  footer: 4
+}
+
+const SPACING_MAP = {
+  // none: '0', Should `none` be an option?
+  condensed: 3,
+  normal: [3, null, null, 4]
+}
+
+const PageLayoutContext = React.createContext<{
+  outerSpacing: keyof typeof SPACING_MAP
+  rowGap: keyof typeof SPACING_MAP
+  columnGap: keyof typeof SPACING_MAP
+}>({
+  outerSpacing: 'normal',
+  rowGap: 'normal',
+  columnGap: 'normal'
+})
+
+// ----------------------------------------------------------------------------
+// PageLayout
+
+export type PageLayoutProps = {
+  /** The maximum width of the page container */
+  containerWidth?: keyof typeof containerWidths
+  /** The spacing between the outer edges of the page container and the viewport */
+  outerSpacing?: keyof typeof SPACING_MAP // Should this be called `padding`?
+  rowGap?: keyof typeof SPACING_MAP
+  columnGap?: keyof typeof SPACING_MAP
+} & SxProp
+
+const containerWidths = {
+  full: '100%',
+  medium: '768px',
+  large: '1012px',
+  xlarge: '1280px'
+}
+
+// TODO: refs
+const Root: React.FC<PageLayoutProps> = ({
+  containerWidth = 'xlarge',
+  outerSpacing = 'normal',
+  rowGap = 'normal',
+  columnGap = 'normal',
+  sx = {},
+  children
+}) => {
+  return (
+    <PageLayoutContext.Provider value={{outerSpacing, rowGap, columnGap}}>
+      <Box sx={merge<BetterSystemStyleObject>({padding: SPACING_MAP[outerSpacing]}, sx)}>
+        <Box
+          sx={{
+            maxWidth: containerWidths[containerWidth],
+            marginX: 'auto',
+            display: 'flex',
+            flexWrap: 'wrap'
+          }}
+        >
+          {children}
+        </Box>
+      </Box>
+    </PageLayoutContext.Provider>
+  )
+}
+
+Root.displayName = 'PageLayout'
+
+// ----------------------------------------------------------------------------
+// Divider (internal)
+
+type DividerProps = {
+  variant?: 'none' | 'line'
+  variantWhenNarrow?: 'inherit' | 'none' | 'line' | 'filled'
+} & SxProp
+
+const horizontalDividerVariants = {
+  none: {
+    display: 'none'
+  },
+  line: {
+    display: 'block',
+    height: 1,
+    backgroundColor: 'border.default'
+  },
+  filled: {
+    display: 'block',
+    height: 8,
+    backgroundColor: 'canvas.inset',
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    boxShadow: (theme: any) =>
+      `inset 0 -1px 0 0 ${theme.colors.border.default}, inset 0 1px 0 0 ${theme.colors.border.default}`
```

Comment:
Unsure about this suggestion: 

How do you feel about typing this with primitives:

`import primitives from '@primer/primitives'`

```suggestion
    boxShadow: ({colors}: {colors: typeof primitives.colors.light}) =>
      `inset 0 -1px 0 0 ${colors.border.default}, inset 0 1px 0 0 ${colors.border.default}`
```

Reply:
Hmm, typing the theme object is a really tricky problem. Casting `theme` to `{colors: typeof primitives.colors.light}` feels like it gives us false confidence when in reality we don't have any type-safety for the theme object. I'm going to give this suggestion a little more thought. I'll probably move forward with `any` for now.